kong/plugins/acl/handler.lua

```diff
@@ -69,8 +69,9 @@ function ACLHandler:access(conf)
     end
 
     if to_be_blocked == false then
-      -- we're allowed, so go and convert 'false' to the header value
-      to_be_blocked = table_concat(consumer_groups, ", ")
+      -- we're allowed, so go and convert 'false' to the header value, if we need it.
+      -- if we don't need it, set a dummy value to save memory for potentially very long strings
+      to_be_blocked = conf.hide_groups_header and nil or table_concat(consumer_groups, ", ")
```

Comment:
no, we cannot set it to `nil`, because the resulting value is cached, and in the cache `nil` means "nothing is cached" (see line 62). The tests all pass, because it will just regenerate every time, without ever caching the results.

You can also leave it at the `false` value it already has. In the example I used an empty string because it gets interned, and because it matches the comments on line 59-60. If you don't use an empty string, then please also update the comments at 59-60.

Reply:
Nice catch. Hadn't looked at this yet, but indeed, this cannot be set to `nil`. Using `false` or `""` is a better fit in this context!

Reply:
Fixed, thanks @Tieske for the further details, I should have guessed you considered nil and knew the implications of it! Back to "" 👍  .